agent/agent_test.go

```diff
@@ -52,12 +52,14 @@ func TestRunnerRun(t *testing.T) {
 
 	t.Run("should return error if extractor could not be found", func(t *testing.T) {
 		proc := mocks.NewProcessor()
+		proc.On("Init", mockCtx, validRecipe.Processors[0].Config).Return(nil).Once()
```

Comment:
why do we need this? IIRC, if extractor could not be found, then processors and sinks Init won't be called.

Reply:
Also on this [test](https://github.com/odpf/meteor/pull/292/files/8b706c97ac7c3a3ef5fb112653dbf7155d5269cf#diff-e12d66df9a328a0b4b4e61abbde9a65eb60a4c8a6a8ca9249a455f173007ac89R133), you need to assert that `RecordRun` are being called instead of plugins and sinks init methods.

Reply:
> why do we need this? IIRC, if extractor could not be found, then processors and sinks Init won't be called.

I changed the flow such that even though extractor initialization fails, it will try to initialization processor and sink. Though it will not try to extract further. The purpose of doing this was to capture all the failing initializations at once and report them to the user.

Do we want this change or shall we step back to the previous one?

Reply:
> Also on this [test](https://github.com/odpf/meteor/pull/292/files/8b706c97ac7c3a3ef5fb112653dbf7155d5269cf#diff-e12d66df9a328a0b4b4e61abbde9a65eb60a4c8a6a8ca9249a455f173007ac89R133), you need to assert that `RecordRun` are being called instead of plugins and sinks init methods.

Yes, I'll add that to the test.